parser/src/fstring.rs

```diff
@@ -130,11 +124,13 @@ impl<'a> FStringParser<'a> {
                             }
                             '{' => {
                                 in_nested = true;
-                                spec_constructor.push(self.expr(ExprKind::Constant {
-                                    value: constant_piece.to_owned().into(),
-                                    kind: None,
-                                }));
-                                constant_piece.clear();
+                                if !constant_piece.is_empty() {
+                                    spec_constructor.push(self.expr(ExprKind::Constant {
+                                        value: constant_piece.to_owned().into(),
+                                        kind: None,
+                                    }));
+                                    constant_piece.clear();
+                                }
```

Comment:
This was another difference I noticed.

In CPython, you get:

```py
>>> import ast
>>> print(ast.dump(ast.parse('f"{foo:{spec}}"')))
Module(body=[Expr(value=JoinedStr(values=[FormattedValue(value=Name(id='foo', ctx=Load()), conversion=-1, format_spec=JoinedStr(values=[FormattedValue(value=Name(id='spec', ctx=Load()), conversion=-1)]))]))], type_ignores=[])
```

Prior to this change, the RustPython AST was similar, except that the inner `FormattedValue` was padded by an empty `Constant` on both sides.